File: Chain of responsability/Python/FrontEnd.py
from InterfaceAyuda import *

class FrontEnd(InterfaceAyuda):
    
    _TIPO_AYUDA = 1

    def __init__(self, s):
        self._sucesor = s

    def getAyuda(self, tipo):

        if(tipo == self._TIPO_AYUDA):
            print("Ayuda desde el FrontEnd")
        else:
            self._sucesor.getAyuda(tipo)        

File: Chain of responsability/Java/FrontEnd.java
public class FrontEnd implements InterfaceAyuda {

    private final int TIPO_AYUDA = 1;
    private InterfaceAyuda sucesor;
   
   FrontEnd(InterfaceAyuda s)
   {
       this.sucesor = s;
   }

    @Override
    public void getAyuda(int tipo) {
        if(tipo == TIPO_AYUDA){
            System.out.println("Ayuda desde el frontEnd");
        } else{
            sucesor.getAyuda(tipo);
        }
    }
    
}
